### PR TITLE
Custom Icons for Clickable Table Rows - RSC-661

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-661

### CODE REVIEW

File: lib/src/base-styles/_nx-tables.scss [hunk 1]
```diff
@@ -306,10 +317,29 @@
...
+@mixin btn-cell-sizing($left-padding, $right-padding) {
+  margin-left: calc(0px - #{$left-padding});
+  margin-right: calc(0px - #{$right-padding});
+  padding-left: $left-padding;
+  padding-right: $right-padding;
+}
+
+.nx-cell__sort-btn, .nx-cell__row-btn {
+  width: min-content;
+
```
Comment (avnvbhatta) [inline lib/src/base-styles/_nx-tables.scss:330]: Should we add a `cursor: pointer;` to this class? Right now, hovering over both the sort cell and the row button icons don't show a pointer, making it seem like both the sort cell and the buttons are not clickable.

https://user-images.githubusercontent.com/7110862/134003686-ed73df59-52a5-482a-bb48-3e8b0e800434.mov

https://user-images.githubusercontent.com/7110862/134003713-25930c2c-b3b6-40ae-ae22-5817907ba2f0.mov

File: gallery/src/components/NxTable/NxTablePage.tsx [hunk 2]
```diff
@@ -22,6 +23,7 @@ import NxTablePaginationFilterExample from './NxTablePaginationFilterExample';
 
 const tableSimpleExampleCode = require('./NxTableSimpleExample?raw');
 const tableClickableExample = require('./NxTableClickableExample?raw');
+const tableClickableCustomExample = require('./NxTableClickableCustomExample?raw');
```
Comment (avnvbhatta) [inline gallery/src/components/NxTable/NxTablePage.tsx:26]: This comment refers to line 239-241

The `clickAccessibleLabel` for `NxTable.Row` needs to be modified. It says "The accessible name to set on the click chevron", but in fact the accessible name could be for any other icon besides a chevron as well.

image

File: lib/src/base-styles/_nx-tables.scss (hunk 1 above)
Comment (rpokorny) [reply lib/src/base-styles/_nx-tables.scss]: > Should we add a cursor: pointer; to this class? Right now, hovering over both the sort cell and the row button icons don't show a pointer, making it seem like both the sort cell and the buttons are not clickable.

Good catch. baa93c79

File: gallery/src/components/NxTable/NxTablePage.tsx (hunk 2 above)
Comment (rpokorny) [reply gallery/src/components/NxTable/NxTablePage.tsx:26]: > This comment refers to line 239-241

The clickAccessibleLabel for NxTable.Row needs to be modified. It says "The accessible name to set on the click chevron", but in fact the accessible name could be for any other icon besides a chevron as well.

71f064f1

**Review by avnvbhatta — APPROVED** [review]

**Pull request merged**